rclpy/rclpy/impl/rmw_implementation_tools.py

```diff
@@ -85,23 +83,8 @@ def import_rmw_implementation():
     if __rmw_implementation_module is not None:
         return __rmw_implementation_module
     available_implementations = get_rmw_implementations()
-    if __selected_rmw_implementation is None:
-        logger = logging.getLogger('rclpy')
-        # prefer FastRTPS, otherwise first in alphabetical order
-        # the same logic is implemented in
-        # rmw_implementation_cmake/cmake/get_default_rmw_implementation.cmake
-        default_impl = 'rmw_fastrtps_cpp'
-        if default_impl in available_implementations:
-            select_rmw_implementation(default_impl)
-        elif available_implementations:
-            select_rmw_implementation(available_implementations[0])  # select the first one
-        else:
-            raise NoImplementationAvailableException()
-        logger.debug("Implicitly selecting the '{0}' rmw implementation."
-                     .format(__selected_rmw_implementation))
-    module_name = '._rclpy__{rmw_implementation}'.format(
-        rmw_implementation=__selected_rmw_implementation,
-    )
+    __selected_rmw_implementation = 'rmw_implementation'
```

Comment:
The PRs are "in progress" because they haven't been CI tested yet and that can only happen after all packages have been converted to not require rmw specific libraries of rcl / rclcpp. Other then that the patches are in the state they will be in for review.

We don't need to build separate Python modules anymore. You will be able to select the rmw implementation the same way as for C++ by setting the environment variable `RMW_IMPLEMENTATION`. There is certainly more cleanup possible since all this rmw impl. selection logic in Python can be removed. But I will very likely leave that to a follow up PR since the complexity of the current set of PRs is already pretty high.